### PR TITLE
[EXPLORER] Fix taskbar icon display problem

### PR DESCRIPTION
**Pull request by TAN-Gaming**
## Purpose

Fix taskbar icon display problem mentioned in [PR #4488](https://github.com/reactos/reactos/pull/4488)

### CODE REVIEW

File: base/shell/explorer/taskswnd.cpp [hunk 1]
```diff
@@ -489,34 +489,28 @@ class CTaskSwitchWnd :
 
     HICON GetWndIcon(HWND hwnd)
     {
-        HICON hIcon = NULL;
+        HICON hIcon;
 #define GET_ICON(type) \
-    SendMessageTimeout(hwnd, WM_GETICON, (type), 0, SMTO_ABORTIFHUNG, 100, (PDWORD_PTR)&hIcon)
+    (HICON)SendMessageW(hwnd, WM_GETICON, (type), 0)
```
Comment (learn-more) [inline base/shell/explorer/taskswnd.cpp:494]: No, we should not hang the shell in case a window hangs!

**Review by learn-more — CHANGES_REQUESTED** [review]
This is incorrect, the timeout is in place so that you do not hang your entire shell on one window that fails to respond.

**Comment by TAN-Gaming** [comment]
Fixed.

**Comment by TAN-Gaming** [comment]
Fixed.

**Comment by TAN-Gaming** [comment]
Fixed.

File: base/shell/explorer/taskswnd.cpp (hunk 1 above)
Comment (TAN-Gaming) [reply base/shell/explorer/taskswnd.cpp]: Fixed.

**Review by HBelusca — APPROVED** [review]

**Comment by binarymaster** [comment]
Would be nice to see the before-after screenshots.

**Comment by TAN-Gaming** [comment]
Before:
https://user-images.githubusercontent.com/86486978/174514435-585d0172-bd73-47bd-b137-611871f25360.mp4

After:
https://user-images.githubusercontent.com/86486978/174514612-87f3b323-d021-4084-bbac-ae4083c710a4.mp4

File: base/shell/explorer/taskswnd.cpp [hunk 2]
```diff
@@ -491,7 +491,7 @@ class CTaskSwitchWnd :
     {
         HICON hIcon = NULL;
 #define GET_ICON(type) \
-    SendMessageTimeout(hwnd, WM_GETICON, (type), 0, SMTO_ABORTIFHUNG, 100, (PDWORD_PTR)&hIcon)
+    SendMessageTimeout(hwnd, WM_GETICON, (type), 0, SMTO_NOTIMEOUTIFNOTHUNG, 100, (PDWORD_PTR)&hIcon)
```
Comment (learn-more) [inline base/shell/explorer/taskswnd.cpp:494]: Is this change required?

Comment (TAN-Gaming) [reply base/shell/explorer/taskswnd.cpp:494]: Yes, this the most significant change that fixes the issue.

**Review by binarymaster — APPROVED** [review]
